pkg/cmd/infra/deployer/deployer.go

```diff
@@ -78,7 +78,7 @@ func NewCommandDeployer(name string) *cobra.Command {
 }
 
 // NewDeployer makes a new Deployer from a kube client.
-func NewDeployer(client kclient.Interface) *Deployer {
+func NewDeployer(client *kclient.Client) *Deployer {
```

Comment:
it can be a second arg to the constructor, just make up your own interface here (I think namespace, name, and PodLogOptions are the only required input), and make an implementation which uses an internal `*kclient.Client` and pass it in